server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeAction.java

```diff
@@ -344,4 +372,244 @@ private static ComponentTreeWsRequest toComponentTreeWsRequest(Request request)
     return wsComponent;
   }
 
+  private ComponentTreeData load(ComponentTreeRequest wsRequest) {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      ComponentDto baseComponent = loadComponent(dbSession, wsRequest);
+      checkPermissions(baseComponent);
+      Optional<SnapshotDto> baseSnapshot = dbClient.snapshotDao().selectLastAnalysisByRootComponentUuid(dbSession, baseComponent.projectUuid());
+      if (!baseSnapshot.isPresent()) {
+        return ComponentTreeData.builder()
+                .setBaseComponent(baseComponent)
+                .build();
+      }
+      Long developerId = searchDeveloperId(dbSession, wsRequest);
+
+      ComponentTreeQuery componentTreeQuery = toComponentTreeQuery(wsRequest, baseComponent);
+      List<ComponentDto> components = searchComponents(dbSession, componentTreeQuery);
+      List<MetricDto> metrics = searchMetrics(dbSession, wsRequest);
+      Table<String, MetricDto, ComponentTreeData.Measure> measuresByComponentUuidAndMetric = searchMeasuresByComponentUuidAndMetric(dbSession, baseComponent, componentTreeQuery,
+              components,
+              metrics, developerId);
+
+      components = filterComponents(components, measuresByComponentUuidAndMetric, metrics, wsRequest);
+      components = sortComponents(components, wsRequest, metrics, measuresByComponentUuidAndMetric);
+
+      int componentCount = components.size();
+      components = paginateComponents(components, wsRequest);
+
+      return ComponentTreeData.builder()
+              .setBaseComponent(baseComponent)
+              .setComponentsFromDb(components)
+              .setComponentCount(componentCount)
+              .setMeasuresByComponentUuidAndMetric(measuresByComponentUuidAndMetric)
+              .setMetrics(metrics)
+              .setPeriods(snapshotToWsPeriods(baseSnapshot.get()))
+              .setReferenceComponentsByUuid(searchReferenceComponentsById(dbSession, components))
+              .build();
+    }
+  }
+
+  private ComponentDto loadComponent(DbSession dbSession, ComponentTreeRequest request) {
+    String componentKey = request.getComponent();
+    String componentId = request.getBaseComponentId();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getBaseComponentId"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)